### PR TITLE
Add MSRV to more lints specified in #6097

### PR DESCRIPTION
**Pull request by Suyash458**
add MSRV to more lints specified in #6097
add instructions for adding msrv in other lints
update tests

 - [x] `redundant_field_names` requires Rust 1.17 due to suggest feature stablized in that version.
 - [x] `redundant_static_lifetimes` requires Rust 1.17 due to suggest feature stablized in that version.
 - [x] `filter_map_next` requires Rust 1.30 due to suggest `Iterator::find_map`.
 - [x] `checked_conversions` requires Rust 1.34 due to suggest `TryFrom`.
 - [x] `match_like_matches_macro` requires Rust 1.42 due to suggest `matches!`. Addressed in #6201
 - [x] `manual_strip` requires Rust 1.45 due to suggest `str::{strip_prefix, strip_suffix}`. Addressed in #6201
 - [x] `option_as_ref_deref` requires Rust 1.40 due to suggest `Option::{as_deref, as_deref_mut}`. Addressed in #6201
 - [x] `manual_non_exhaustive` requires Rust 1.40 due to suggest `#[non_exhaustive]`. Addressed in #6201
 - [x] `manual_range_contains` requires Rust 1.35 due to suggest `Range*::contains`.
 - [x] `use_self` requires Rust 1.37 due to suggest `Self::Variant on enum`.
 - [x] `mem_replace_with_default` requires Rust 1.40 due to suggest `mem::take`.
 - [x] `map_unwrap_or` requires Rust 1.41 due to suggest `Result::{map_or, map_or_else}`.
 - [x] `missing_const_for_fn` requires Rust 1.46 due to `match/if/loop in const fn` needs that version.

changelog: Add MSRV config to more lints. ^This is now the complete list, AFAWK

### CODE REVIEW

**Comment by flip1995** [comment]
LGTM. There are some extra blank lines added (mostly before the `if meets_msrv` conditions. 

Do you want to add the rest of the lints in this PR or split this up?

**Comment by Suyash458** [comment]
@flip1995 think I'll add all the lints in this one along with some instructions for introducing `msrv` for other lints. I'll run `cargo dev fmt` before I finish this up

**Comment by flip1995** [comment]
Great! Please ping me, when you done, so I don't miss it

**Comment by Suyash458** [comment]
@flip1995 could you please review these changes?

File: tests/ui/min_rust_version_attr.stderr [hunk 1]
```diff
@@ -1,37 +1,28 @@
...
-note: the prefix was tested here
-  --> $DIR/min_rust_version_attr.rs:59:9
-   |
-LL |         if s.starts_with("hello, ") {
-   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-help: try using the `strip_prefix` method
-   |
-LL |         if let Some(<stripped>) = s.strip_prefix("hello, ") {
-LL |             assert_eq!(<stripped>.to_uppercase(), "WORLD!");
+error[E0425]: cannot find value `x` in this scope
```
Comment (flip1995) [inline tests/ui/min_rust_version_attr.stderr:1]: I don't think this test should have any output. These are all Rust errors.

File: tests/ui/min_rust_version_attr.rs [hunk 2]
```diff
@@ -42,12 +42,84 @@ pub fn manual_strip_msrv() {
...
+    redundant_static_lifetime();
     option_as_ref_deref();
     match_like_matches();
     match_same_arms();
     match_same_arms2();
     manual_strip_msrv();
+    manual_range_contains();
+    use_self();
+    replace_with_default();
+    map_unwrap_or();
```
Comment (flip1995) [inline tests/ui/min_rust_version_attr.rs:122]: I think you're missing a test for `missing_const_for_fn`?

Comment (Suyash458) [reply tests/ui/min_rust_version_attr.rs:131]: whoops, missed that one. I've added it now

File: tests/ui/min_rust_version_attr.stderr (hunk 1 above)
Comment (Suyash458) [reply tests/ui/min_rust_version_attr.stderr]: Yes, I've removed these errors

File: tests/ui/min_rust_version_attr.stderr [hunk 3]
```diff
@@ -1,12 +1,12 @@
 error: stripping a prefix manually
-  --> $DIR/min_rust_version_attr.rs:60:24
+  --> $DIR/min_rust_version_attr.rs:142:24
    |
 LL |             assert_eq!(s["hello, ".len()..].to_uppercase(), "WORLD!");
    |                        ^^^^^^^^^^^^^^^^^^^^
    |
    = note: `-D clippy::manual-strip` implied by `-D warnings`
```
Comment (flip1995) [inline tests/ui/min_rust_version_attr.stderr:7]: Wait, `manual_strip` should not lint here?

**Review by flip1995 — COMMENTED** [review]
LGTM, but something is going on with `manual_strip`

File: tests/ui/min_rust_version_attr.stderr (hunk 3 above)
Comment (Suyash458) [reply tests/ui/min_rust_version_attr.stderr:7]: This is from the test that checks for the equality of `msrv`

```rust
mod meets_msrv {
    #![feature(custom_inner_attributes)]
    #![clippy::msrv = "1.45.0"]

    fn main() {
        let s = "hello, world!";
        if s.starts_with("hello, ") {
            assert_eq!(s["hello, ".len()..].to_uppercase(), "WORLD!");
        }
    }
}
```

**Review by flip1995 — APPROVED** [review]
Thanks! Can you squash some of the commits, please? There are 3 or 4 formatting commits.

**Comment by Suyash458** [comment]
> 
> 
> Thanks! Can you squash some of the commits, please? There are 3 or 4 formatting commits.

Just did! Are we good to merge this?

**Comment by flip1995** [comment]
Thanks! I think we can start writing up the internals blog post :+1: 

@bors r+

**Pull request merged**